packages/arb-node-core/staker/staker.go

```diff
@@ -114,7 +122,59 @@ func (s *Staker) RunInBackground(ctx context.Context) chan bool {
 	return done
 }
 
+func (s *Staker) shouldAct(ctx context.Context) bool {
+	var gasPriceHigh = false
+	var gasPriceFloat float64
+	gasPrice, err := s.client.SuggestGasPrice(ctx)
+	if err != nil {
+		logger.Warn().Err(err).Msg("error getting gas price")
+	} else {
+		gasPriceFloat = float64(gasPrice.Int64()) / 1e9
+		if gasPriceFloat >= s.config.L1PostingStrategy.HighGasThreshold {
+			gasPriceHigh = true
+		}
+	}
+	latestBlockInfo, err := s.client.BlockInfoByNumber(ctx, nil)
+	if err != nil {
+		logger.Warn().Err(err).Msg("error getting latest block")
+		return true
+	}
+	latestBlockNum := latestBlockInfo.Number.ToInt()
+	if s.lastActCalledBlock == nil {
+		s.lastActCalledBlock = latestBlockNum
```

Comment:
Seems like `lastActCalledBlock` should also be updated when the validator takes an action, but I don't think it does currently

Reply:
I completely missed this, thanks

Reply:
Specifically, it should be updated not only when the validator takes an action, but also when it doesn't, as the highGasBlocksBuffer is still updated. I've fixed this.